### PR TITLE
Semantic table can now answer type questions from identifier name.

### PR DESCRIPTION
**Pull request by Mats-SX**
This works since the semantic table is guaranteed to contain no conflicting type information for a given identifier name, regardless of input position.

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/QueryPlanner.scala [hunk 1]
```diff
@@ -28,9 +28,9 @@ trait QueryPlanner {
   def plan(plannerQuery: UnionQuery)(implicit context: LogicalPlanningContext): LogicalPlan
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/QueryPlanner.scala:28]: Does these changes belong to this PR?

Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/QueryPlanner.scala:28]: Well, not really from the branch name or the PR name, but me and Andrés felt that these changes were so small they might as well be part of this PR since it was also pretty small and isolated. Do you think we should make a dedicated PR instead?

Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/QueryPlanner.scala:28]: No that's fine, just wanted to be sure it was not added here by mistake
